routing/index_graph.hpp

```diff
@@ -77,16 +119,154 @@ class IndexGraph final
+  void ApplyRestrictionOnly(RestrictionPoint const & restrictionPoint);
+
+  void ApplyRestrictionOnlyRealFeatures(RestrictionPoint const & restrictionPoint);
+  void ApplyRestrictionNoRealFeatures(RestrictionPoint const & restrictionPoint);
+
+  /// \brief Add restrictions in |restrictions| to |m_ftPointIndex|.
+  void ApplyRestrictions(RestrictionVec const & restrictions);
+
+  /// \brief Fills |singleFeaturePath| with points from point |from| to point |to|
+  /// \note |from| and |to| has to belong the same feature.
+  /// \note The order on points in items of |connectionPaths| is from |from| to |to|.
+  void GetSingleFeaturePaths(RoadPoint from, RoadPoint to, vector<RoadPoint> & singleFeaturePath);
+
+  /// \brief  Fills |connectionPaths| with all path from joint |from| to joint |to|.
+  /// If |from| and |to| don't belong to the same feature |connectionPaths| an exception
+  /// |RootException| with be raised.
+  /// If |from| and |to| belong to only one feature |connectionPaths| will have one item.
+  /// It's most common case.
+  /// If |from| and |to| could be connected by several feature |connectionPaths|
+  /// will have several items.
+  /// \note The order on points in items of |connectionPaths| is from |from| to |to|.
+  void GetConnectionPaths(Joint::Id from, Joint::Id to,
+                          vector<vector<RoadPoint>> & connectionPaths);
+
+  /// \brief Fills |shortestConnectionPath| with shortest path from joint |from| to joint |to|.
+  /// \note The order on points in |shortestConnectionPath| is from |from| to |to|.
+  /// \note In most cases the method doesn't load geometry to find the shortest path
+  /// because there's only one path between |from| and |to|.
+  /// \note if |shortestConnectionPath| could be filled after call of the method
+  /// an exception of type |RootException| is raised. It could happend,
+  /// for example, if |from| and |to| are connected with a pedestrian road
+  /// but a car route is looked for.
+  void GetShortestConnectionPath(Joint::Id from, Joint::Id to,
+                                 vector<RoadPoint> & shortestConnectionPath);
+
+  /// \brief Fills |featureConnectionPath| with a path from |from| to |to| of points of |featureId|.
+  void GetFeatureConnectionPath(Joint::Id from, Joint::Id to, uint32_t featureId,
+                                vector<RoadPoint> & featureConnectionPath);
+
+  void GetOutgoingGeomEdges(vector<JointEdge> const & outgoingEdges, Joint::Id center,
+                            vector<JointEdgeGeom> & outgoingGeomEdges);
+
+  void CreateFakeFeatureGeometry(vector<RoadPoint> const & geometrySource, RoadGeometry & geometry);
+
+  /// \returns RoadGeometry by a real or fake featureId.
+  RoadGeometry const & GetRoad(uint32_t featureId);
+
+  bool IsFakeFeature(uint32_t featureId) const { return featureId >= kStartFakeFeatureIds; }
+
+  static uint32_t const kStartFakeFeatureIds = 1024 * 1024 * 1024;
```

Comment:
Please, move this to the beginning of the public section. https://google.github.io/styleguide/cppguide.html#Declaration_Order